client/src/entrypoints/admin/page-editor.js

```diff
@@ -402,6 +384,122 @@ window.updateFooterSaveWarning = (formDirty, commentsDirty) => {
   }
 };
 
+function initPreview() {
+  const previewPanel = document.querySelector('.preview-panel');
+  // Preview panel is not shown if the page does not have any preview modes
+  if (!previewPanel) return;
+
+  const previewButtons = previewPanel.querySelectorAll('[data-preview-size]');
+
+  const togglePreviewSize = (event) => {
+    const currentButton = event.currentTarget;
+    const size = currentButton.dataset.previewSize;
+    const hasErrors = previewPanel.classList.contains(
+      'preview-panel--has-errors',
+    );
+
+    previewButtons.forEach((b) => b.setAttribute('aria-current', 'false'));
+    currentButton.setAttribute('aria-current', 'true');
+    previewPanel.dataset.activePreviewSize = size;
+    previewPanel.className = `preview-panel preview-panel--${size}`;
```

Comment:
That was why I had the (unused) `data-active-preview-size` attribute, and I used it in the CSS to select the size correctly, but then I changed it because it's against our guidelines to use data attributes in CSS.